csrc/cpu/cpu_types.hpp

```diff
@@ -123,6 +124,34 @@ struct BF16Vec32 : public Vec<BF16Vec32> {
 
   void save(void *ptr) const { *reinterpret_cast<__m512i *>(ptr) = reg; }
 };
+#else
```

Comment:
I would prefer an explicit check for AVX2 and then having an #else section that is a static assert about unsupported ISA

Reply:
Thanks @mgoin for the review.

We already have the avx2 check in `cpu_extension.cmake`.
It's also fine to check it again here.

However, the original AVX512 implementation also requires avx2.
So I would suggest checking it at the top of the file (please see the second commit).

What do you think?
Thanks.

Reply:
Hi @mgoin , are you also fine with this change?
Thanks.

Reply:
I'm fine with this. I was just thinking towards being explicit with what each case was, rather than having the assumption be that "else" was avx2, since other ISAs could be added in the future. This is a fine state for now

Reply:
> I'm fine with this. I was just thinking towards being explicit with what each case was, rather than having the assumption be that "else" was avx2, since other ISAs could be added in the future. This is a fine state for now

Thanks @mgoin .
And I need your help to merge it.
Thanks.